modules/core/src/main/scala/org/scalasteward/core/repoconfig/RepoConfigAlg.scala

```diff
@@ -16,34 +16,49 @@
 
 package org.scalasteward.core.repoconfig
 
+import better.files.File
 import cats.data.OptionT
 import cats.implicits._
 import io.chrisdavenport.log4cats.Logger
 import io.circe.config.parser
+import org.scalasteward.core.application.Config
 import org.scalasteward.core.data.Update
 import org.scalasteward.core.io.{FileAlg, WorkspaceAlg}
 import org.scalasteward.core.repoconfig.RepoConfigAlg._
 import org.scalasteward.core.util.MonadThrowable
 import org.scalasteward.core.vcs.data.Repo
 
 final class RepoConfigAlg[F[_]](implicit
+    config: Config,
     fileAlg: FileAlg[F],
     logger: Logger[F],
     workspaceAlg: WorkspaceAlg[F],
     F: MonadThrowable[F]
 ) {
   def readRepoConfigOrDefault(repo: Repo): F[RepoConfig] =
-    readRepoConfig(repo).map(_.getOrElse(RepoConfig.default))
+    readRepoConfig(repo).flatMap { config =>
+      config.map(F.pure).getOrElse(defaultRepoConfig())
+    }
+
+  /**
+    * Default configuration will try to read file specified in config.reposDefaultConfig first; if not found - fallback to empty configuration.
+    * Note it's not lazy since we want to get config updates if you decide to change config in the middle of the process
+    */
+  def defaultRepoConfig(): F[RepoConfig] =
```

Comment:
> Note it's not lazy since we want to get config updates if you decide to change config in the middle of the process

I can move it to be val - just thinking if you have 100+ repos and you found an issue or smth, it will give you hot reload

wdyt?

Reply:
Since the type is still `F[RepoConfig]` it will not be eagerly evaluated if this is changed to a `val`.

Reply:
you mean even if I leave it as is it won't be eager even if it's `def`?

Reply:
ok set it to val for now

Reply:
I mean that it doesn't semantically matter if this is a `def` or `val` because the effects of that statement are not executed when it is evaluated but when the `F` is run.